src/main/java/com/marklogic/client/ext/document/ContentIdExtractor.java

```diff
@@ -1,9 +1,13 @@
 package com.marklogic.client.ext.document;
 
+import java.io.IOException;
+
 import com.marklogic.client.io.marker.AbstractWriteHandle;
 
 public interface ContentIdExtractor {
 
 	String extractId(AbstractWriteHandle content);
-	
-}
+	String extractId(AbstractWriteHandle content, String path);
+	String extractId(AbstractWriteHandle content, String[] path);
+	String extractId(AbstractWriteHandle content, String topic, Integer partition, long offset, String idStrategy);
```

Comment:
Added. Test  methods in `ConvertSinkRecordTest `for different ID strategies. Pls take a look. The `ContentIdExtractor `is not a functional interface any more, because there are more than one abstract methods. So, the current junit method using lambda had to be refactored. Pls see the changes in `ConvertSinkR.ecordTest.noPropertiesSet` 
Also note that build.gradle should have `testCompile "com.google.code.gson:gson:2.8.6"` because real JSONs should be built for some ID strategies.

Reply:
Looks good! Thanks.